### ISSUE TITLE
Extend "Plugin best practices" doc with thoughts on easy extendability

### ISSUE THREAD

**Issue by sebastiangreger** [getkirby/getkirby.com#1009]
Not sure is this an "idea" issue, or should it be a issue (or PR) in the getkirby.com repo?

When writing a plugin with snippets, these can be overridden by a site using a snippet of the same name as registered. In order to allow developers to keep their `snippets` folder clean, it would be good practice to not register them with

```
'snippets'     => [
    'pluginname_snippetname' => __DIR__ . '/snippets/snippetname.php',
]
```

but

```
'snippets'     => [
    'pluginname/snippetname' => __DIR__ . '/snippets/snippetname.php',
]
```

instead. This allows to place plugin-related custom snippets into folder `site/snippets/pluginname` rather than adding them to the main `site/snippets` folder. I got [feedback](https://github.com/sebastiangreger/kirby3-commentions/issues/78) on that related to one of my plugins, and today - now sensitized to it - encountered the same while implementing somebody else's plugin.

A recommendation to use the latter style in the "Best practices" document could help establish this as a pattern? Maybe as a subchapter along with some other "best practices" that make it easy to override plugin defaults for site authors?

**Comment by lukasbestle** [comment]
Good point, I fully agree that this is a best-practice worth sharing. 👍

I have moved the issue to the getkirby.com repo.
